smallrye-reactive-messaging-pulsar/src/main/java/io/smallrye/reactive/messaging/pulsar/PulsarOutgoingChannel.java

```
@@ -109,7 +109,8 @@ private Uni<Void> sendMessage(Message<?> message) {
                         OutgoingMessageMetadata.setResultOnMessage(message, mid);
                         return Uni.createFrom().completionStage(message.ack());
                     } else {
-                        return Uni.createFrom().completionStage(message.nack(t));
+                        return Uni.createFrom().completionStage(message.nack(t))
+                                .chain(() -> Uni.createFrom().failure(t));
```

Maybe there is a better solution to do that, I've noticed that the failure reporting is already in the caller, so the intention is to propagate the error instead of silently handling it here.

---

OK, just noticed that this one is to be reverted, as the Nack will affect the health checks, could it be possible to add logging here?

---

Yes, logging would be great!

---

I'll propose something.